eng/tools/versioning/set-dev.js

```diff
@@ -23,55 +23,196 @@ const path = require("path");
+  rushPackages = updateDependencySection(
+    rushPackages,
+    rushPackages[package].json.dependencies,
+    buildId
   );
 
-  console.log(appropriateVersionedPackages);
+  console.log("checking devDependencies ..");
+  rushPackages = updateDependencySection(
+    rushPackages,
+    rushPackages[package].json.devDependencies,
+    buildId
+  );
 
-  let targetPackages = appropriateVersionedPackages.map(project =>
-    path.resolve(path.join(repoRoot, project.projectFolder))
+  console.log("checking peerDependencies ..");
+  rushPackages = updateDependencySection(
+    rushPackages,
+    rushPackages[package].json.peerDependencies,
+    buildId
   );
 
-  for (const targetFolder of targetPackages) {
-    const targetPackageJson = path.join(targetFolder, "package.json");
-    const packageJsonContents = await versionUtils.readFileJson(
-      targetPackageJson
-    );
-    const newVersion = getDevVersion(packageJsonContents.version, buildId);
-    console.log(`File ${targetPackageJson} version updated to ${newVersion}`);
-    const updatedPackageSpec = {
-      ...packageJsonContents,
-      version: newVersion
-    };
-    await versionUtils.writePackageJson(targetPackageJson, updatedPackageSpec);
-
-    await versionUtils.updatePackageConstants(
-      targetFolder,
-      packageJsonContents,
-      newVersion
+  return rushPackages;
+};
+
+const makeDependencySectionConsistentForPackage = (rushPackages, dependencySection, depName) => {
+  console.log("Inside make dep");
+  if (dependencySection && dependencySection[depName]) {
+    depVersionRange = dependencySection[depName];
+
+    console.log(`checking ${depName}:${depVersionRange}...`);
+
+    // If the dependency isn't part of the Rush workspace, skip it
+    if (!rushPackages[depName]) {
+      return rushPackages;
+    }
+
+    // Compare the dependency version range with the package's current version
+    const packageVersion = rushPackages[depName].json.version;
+
+    console.log(`version in package's dep = ${depVersionRange}`);
+    console.log(`dep's version = ${packageVersion}`);
+    const parsedPackageVersion = semver.parse(packageVersion);
+    const parsedDepMinVersion = semver.minVersion(depVersionRange);
+    // If the dependency range is satisfied by the package's current version,
+    // replace it with an exact match to the package's new version
+    if (
+      parsedDepMinVersion.major == parsedPackageVersion.major &&
+      parsedDepMinVersion.minor == parsedPackageVersion.minor &&
+      parsedDepMinVersion.patch == parsedPackageVersion.patch &&
+      rushPackages[depName].newVer !== undefined
+    ) {
+      // Commit the dependency version to the JSON document
+      dependencySection[
+        depName
+      ] = `^${parsedPackageVersion.major}.${parsedPackageVersion.minor}.${parsedPackageVersion.patch}-dev`;
+    }
+  }
+  return rushPackages;
+};
+
+const makeInternalDependenciesConsistent = (rushPackages, package,depName) => {
```

Comment:
Can we name this `updateOtherProjectDependencySections`

Reply:
done